src/DGtal/topology/Object.ih

```diff
@@ -60,7 +61,12 @@
 template <typename TDigitalTopology, typename TDigitalSet>
 inline
 DGtal::Object<TDigitalTopology, TDigitalSet>::Object()
-  : myTopo( 0 ), myPointSet( 0 ), myConnectedness( UNKNOWN )
+  : myTopo( nullptr ),
```

Comment:
Yes I checked also. According to me, it is for mostly for the auto keyword, Overloading with int and pointer is kind of a weird usage.